### PR TITLE
Add support for anonymousReadOnly in LdapProperties

### PR DESCRIPTION
**Pull request by eddumelendez**
See gh-11722

### CODE REVIEW

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ldap/LdapProperties.java [hunk 1]
```diff
@@ -60,6 +60,11 @@
 	 */
 	private Map<String, String> baseEnvironment = new HashMap<>();
 
+	/**
+	 * Enable anonymous read only.
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ldap/LdapProperties.java]: This description could use some love, something like

> Whether read-only operations should use an anonymous environment.

File: spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/ldap/LdapAutoConfigurationTests.java [hunk 2]
```diff
@@ -16,13 +16,12 @@
...
 package org.springframework.boot.autoconfigure.ldap;
 
-import org.junit.After;
 import org.junit.Test;
 
-import org.springframework.boot.autoconfigure.context.PropertyPlaceholderAutoConfiguration;
-import org.springframework.boot.test.util.TestPropertyValues;
-import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/ldap/LdapAutoConfigurationTests.java]: Can the refactoring of the test be done in a separate commit please? First that with the test unchanged and then an extra commit on top with the feature.

**Comment by snicoll** [comment]
Thanks @eddumelendez - regarding the test refactoring, that's not what I had in mind by "separate commit". What I meant here is first a commit that upgrades the existing tests and then one commit  on  top that implements the feature (with the relevant tests). I've restructured your PR with that and I've also [polished it](https://github.com/spring-projects/spring-boot/commit/960989cfe60e447229af954c31846ecb3b1f0461)

**Comment by eddumelendez** [comment]
@snicoll I'll take it into account next time. Thanks! :+1:
